specs/GetFavicon.md

```diff
@@ -0,0 +1,149 @@
+                CHECK_FAILURE(sender->get_FaviconUri(&url));
+                wil::com_ptr<IStream> pStream = SHCreateMemStream(nullptr, 0);
+
+                sender->GetFavicon(
+                    COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG, 
+                    pStream,
+                    Callback<
+                        ICoreWebView2GetFaviconCompletedHandler>(
+                        [&pStream, this](HRESULT code) -> HRESULT {
+                                    Gdiplus::Bitmap* pBitmap = new Gdiplus::Bitmap(pStream);
+                                    HICON icon;
+                                    pBitmap->GetHICON(&icon);
+                                    SendMessage(
+                                        m_appWindow->GetMainWindow(), WM_SETICON, ICON_SMALL,
+                                        (LPARAM)icon);
+                            return S_OK;
+                        })
+                        .Get());
+
+                return S_OK;
+            })
+            .Get(),
+        &m_faviconChangedToken));  
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    System.IO.Stream stream = new System.IO.MemoryStream();
+    await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png,
+        stream);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+If a Web page does not have a Favicon, then the event is only fires when the
+page is navigated to. The Favicon would be an empty image and Uri. The
+user is expected to handle this scenario.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// This interface is a handler for when the `Favicon` is changed.
+/// The sender is the embedded_browser which handle the change and the
+/// second argument is null.
+/// For more information see `add_FaviconChanged`.
+[uuid(2913DA94-833D-4DE0-8DCA-900FC524A1A4), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalFaviconChangedEventHandler : IUnknown {
+  /// Called to notify the favicon changed.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+/// This interface is a handler for the completion of the copying for the`imageStream`.
+/// The 'error_code` is E_NOT_SET if the there is no image. Otherwise error_code 
+/// is the result from the image write operation.
+/// For more details, see the `GetFavicon` API.
+[uuid(A2508329-7DA8-49D7-8C05-FA125E4AEE8D), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalGetFaviconCompletedHandler : IUnknown {
+  /// Called to notify the favicon has been retrieved.
+  HRESULT Invoke([in] HRESULT error_code);
+}
+
+/// This is the ICoreWebView2 Experimental Favicon interface.
+/// It has the capability to fire an event for the changing of the
```

Comment:
Would be good to say which favicon we're talking about. Also we use the word 'raise' instead of 'fire' for events. Would be nice to include a link to MDN favicon definition.

```suggestion
/// The `FaviconChanged` event is raised when the [favicon](https://developer.mozilla.org/en-US/docs/Glossary/Favicon) of the top-level document changes. The favicon can change when navigating to a new document or if script dynamically changes the favicon. When navigating from a document with a favicon to a new URI, the FaviconChanged event will be raised first for first navigating to a new document which declares a favicon in its HTML or has script to set its favicon, the document and the FaviconChanged event will be raised for that
```